src/main/java/com/google/devtools/build/lib/runtime/BuildSummaryStatsModule.java

```diff
@@ -228,6 +251,41 @@ public void buildComplete(BuildCompleteEvent event) {
         criticalPathComputer = null;
       }
       profilePath = null;
+      cpuUserTimeForActions = Duration.ofMillis(0);
+      cpuSystemTimeForActions = Duration.ofMillis(0);
+      cpuTimeForBazelJvm = Duration.ofMillis(UNKNOWN_CPU_TIME);
+    }
+  }
+
+  private static String formatCpuTime(long milliseconds) {
+    if (milliseconds == UNKNOWN_CPU_TIME) {
+      return "???s";
+    } else {
+      return String.format("%.2fs", milliseconds / 1000.0);
     }
   }
+
+  //If anyone of the CPU time(cpuUserTimeActions, cpuSystemTimeActions and cpuTimeBazelJvm) is UNKNOWN, then the total CPU time become UNKNOWN.
+  private static long sumCpuTimes(long cpuUserTimeActions, long cpuSystemTimeActions, long cpuTimeBazelJvm) {
+    if ((cpuUserTimeActions == UNKNOWN_CPU_TIME) || (cpuSystemTimeActions == UNKNOWN_CPU_TIME) || (cpuTimeBazelJvm == UNKNOWN_CPU_TIME)) {
+      return UNKNOWN_CPU_TIME;
+    } else {
+      return cpuUserTimeActions + cpuSystemTimeActions + cpuTimeBazelJvm;
+    }
+  }
+  
+  private static Duration addCPUTime(Optional<Duration> sumDuration, Duration termDuration) {
```

Comment:
This method needs to be threadsafe, since it's called from a method marked `@AllowConcurrentEvents`.